crates/driver/src/domain/mempools.rs

```diff
@@ -266,10 +269,14 @@ pub enum Error {
     #[error("Mined reverted transaction: {tx_id:?}, block number: {block_number}")]
     Revert {
         tx_id: eth::TxId,
+        submitted_at_block: BlockNo,
         block_number: BlockNo,
     },
     #[error("Simulation started reverting during submission, block number: {block_number}")]
-    SimulationRevert { block_number: BlockNo },
+    SimulationRevert {
+        submitted_at_block: BlockNo,
+        block_number: BlockNo,
```

Comment:
Unless this causes backwards compatibility issues, this field should probably be named more expressively "reverted at block"